src/sentry/workflow_engine/models/detector.py

```diff
@@ -105,3 +102,12 @@ def detector_handler(self) -> DetectorHandler | None:
     def get_audit_log_data(self) -> dict[str, Any]:
         # TODO: Create proper audit log data for the detector, group and conditions
         return {}
+
+
+@receiver(pre_save, sender=Detector)
+def enforce_config_schema(sender, instance: Detector, **kwargs):
+    group_type = instance.group_type
+    if not group_type:
+        raise ValueError(f"No group type found with type {instance.type}")
+    config_schema = group_type.detector_config_schema
+    instance.validate_config(config_schema)
```

Comment:
We should probably also check that `detector_config_schema` is set too

Reply:
this is set to an empty dict by default in GroupType, do we need anything else?

Reply:
Oh, I guess that's fine... One problem with defaulting it to an empty dict is that if people forget to add it then they won't get an error, but this works as is